### PR TITLE
Release v2.0.0

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
This is mainly a compatibility release for the major preCICE release v3.0.0 https://github.com/precice/precice/releases/tag/v3.0.0. This especially breaks compatibility of the FEniCS adapter with older preCICE versions and is therefore considered as a major release of the FEniCS adapter.

The remaining changes are mainly minor bugfixes and documentation updates.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -39,6 +39,6 @@
       author_email='info@precice.org',
       license='LGPL-3.0',
       packages=['fenicsprecice'],
-      install_requires=['pyprecice>=2.0.0', 'scipy', 'numpy>=1.13.3', 'mpi4py'],
+      install_requires=['pyprecice>=3.0.0.0.dev0', 'scipy', 'numpy>=1.13.3', 'mpi4py'],
```
Comment (IshaanDesai) [inline setup.py:42]:
```suggestion
      install_requires=['pyprecice>=3.0.0.0', 'scipy', 'numpy>=1.13.3', 'mpi4py'],
```

**Review by IshaanDesai — COMMENTED** [review]
Looks good :+1: I did not test any cases specifically, but the CI checks should be sufficient. pyprecice version used should be v3.0.0.0.

**Comment by uekerman** [comment]
> This especially breaks compatibility of the FEniCS adapter with older preCICE versions and is therefore considered as a major release of the FEniCS adapter.

Normally, this is no reason for a major release. In the end, preCICE is only a dependency.

**Comment by IshaanDesai** [comment]
> Normally, this is no reason for a major release. In the end, preCICE is only a dependency.

But a FEniCS user calls the adapter API in their code, so their code will break when we release. This is fundamentally different than how the OpenFOAM adapter is used. Does this not warrant a major release?

File: setup.py (hunk 1 above)
Comment (BenjaminRodenberg) [reply setup.py:42]: Aaaaah. Very good :)

**Comment by BenjaminRodenberg** [comment]
@uekerman for the FEniCS adapter we have to do a breaking release, if we want to support preCICE v3. See https://github.com/precice/fenics-adapter/pull/153 and how the API of the adapter changes. We also had to updates all the tutorials. See https://github.com/precice/tutorials/commit/63273dd86ce2bf5ee5df50bc7ebb8e3e5310c1a7.

File: .github/workflows/build-docker.yml [hunk 2]
```diff
@@ -13,10 +21,28 @@ jobs:
...
+        run: |
+          if [[ '${{ env.ADAPTER_REF }}' == 'master' ]]; then
+              echo "PYTHON_BINDINGS_REF=latest" >> "$GITHUB_ENV"
+              echo "TAG=latest" >> "$GITHUB_ENV"
+              echo "Building TAG: latest"
+          else
+              echo "PYTHON_BINDINGS_REF=${{ env.ADAPTER_REF }}" >> "$GITHUB_ENV"
+              echo "TAG=${{ env.ADAPTER_REF }}" >> "$GITHUB_ENV"
+              echo "Building TAG: ${{ env.ADAPTER_REF }}"
+          fi
```
[the remark this answers is not in the document]
Comment (IshaanDesai) [reply .github/workflows/build-docker.yml:45]: No solution yet. This workflow will break on master, and then work again when we merge master to develop. So we will have the image, just from develop.

**Review by IshaanDesai — APPROVED** [review]

File: .github/workflows/build-docker.yml [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
+      branch:
+        type: choice
+        description: branch to build the container from
+        options:
+          - develop
+          - master     
   push:
     branches:
       - develop
+      - master
```
[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply .github/workflows/build-docker.yml:15]: I checked the dockerfile and things look ok. I think I will risk it :crossed_fingers:

File: .github/workflows/build-docker.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (BenjaminRodenberg) [reply .github/workflows/build-docker.yml:45]: Ok. Then let's do a manual docker release, if necessary.

**Pull request merged**